docs/source-2.0/additional-specs/rules-engine/standard-library.rst

```diff
@@ -38,6 +38,53 @@ parameter is equal to the value ``false``:
     }
 
 
+.. _rules-engine-standard-library-coalesce:
+
+``coalesce`` function
+=====================
+
+Summary
+    Evaluates arguments in order and returns the first non-empty result, otherwise returns the result of the last
```

Comment:
We should make sure that we have test cases for implementors that cover coalesce with boolean types to ensure that handling of False values is consistent.

Reply:
Where would such a test be added?

Reply:
I think we've generally added them under the test rule files, eg: [parse-url.smithy](https://github.com/smithy-lang/smithy/blob/main/smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/parse-url.smithy) as well as in internal documentation.